platform/src/apis/Platform.Api.Insight/Census/CensusService.cs

```diff
@@ -27,6 +29,52 @@ public async Task<IEnumerable<CensusHistoryModel>> GetHistoryAsync(string urn)
         return await conn.QueryAsync<CensusHistoryModel>(sql, parameters);
     }
 
+    public async Task<IEnumerable<CensusHistoryModel>> GetHistoryAvgComparatorSetAsync(string urn, CensusParameters queryParams)
+    {
+        var parameters = new
+        {
+            URN = urn
+        };
+
+        var sourceName = queryParams.Dimension switch
+        {
+            CensusDimensions.Total => "SchoolCensusAvgComparatorSet",
+            CensusDimensions.HeadcountPerFte => "SchoolCensusAvgPerFteComparatorSet",
+            CensusDimensions.PercentWorkforce => "SchoolCensusAvgPercentageOfWorkforceFteComparatorSet",
+            CensusDimensions.PupilsPerStaffRole => "SchoolCensusAvgPupilsPerStaffComparatorSet",
+            _ => throw new ArgumentOutOfRangeException(nameof(queryParams))
```

Comment:
[nitpick] I think this should be `nameof(queryParams.Dimension)` or similar. Same below, and possibly in the Expenditure service.

Reply:
I think this is better as `nameof(dimension)` and alter the parameters for these methods. I don't think passing the whole queryParams as an argument actually makes that much sense. Amended in the latest commits